### PR TITLE
Fcom 211

### PR DESCRIPTION
**Pull request by hodorsoft**
.netstandard + csv & file output support (hox, breaking changes due to task settings arragement)

### CODE REVIEW

File: Frends.Community.Oracle.Query/Extensions.cs [hunk 1]
```diff
@@ -16,35 +20,230 @@ public static TEnum ConvertEnum<TEnum>(this Enum source)
...
+            }
+        }
+
+        /// <summary>
+        /// Write query results to xml string or to a file
+        /// </summary>
+        /// <param name="command"></param>
+        /// <param name="output"></param>
+        /// <returns></returns>
+        public static string ToXml(this OracleCommand command, OutputProperties output)
```
Comment (narnil) [inline Frends.Community.Oracle.Query/Extensions.cs]: This is not used anywhere and should be removed.

File: Frends.Community.Oracle.Query/Frends.Community.Oracle.Query.csproj [hunk 2]
```diff
@@ -1,63 +1,31 @@
...
-    <DebugType>pdbonly</DebugType>
-    <Optimize>true</Optimize>
-    <OutputPath>bin\Release\</OutputPath>
-    <DefineConstants>TRACE</DefineConstants>
-    <ErrorReport>prompt</ErrorReport>
-    <WarningLevel>4</WarningLevel>
-    <DocumentationFile>bin\Release\Frends.Community.Oracle.Query.xml</DocumentationFile>
+
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|AnyCPU'">
+    <DocumentationFile>C:\Users\raustvi\Documents\duunit\Frends.Community.Tasks\Frends.Community.Oracle.Query\Frends.Community.Oracle.Query\Frends.Community.Oracle.Query.xml</DocumentationFile>
```
Comment (narnil) [inline Frends.Community.Oracle.Query/Frends.Community.Oracle.Query.csproj]: Hard coded value pointing to user folder should be changed to use project folder dynamically.

File: Frends.Community.Oracle.Query/Extensions.cs [hunk 3]
```diff
@@ -16,35 +20,230 @@ public static TEnum ConvertEnum<TEnum>(this Enum source)
...
+
+                // create json result
+                using (var fileWriter = output.OutputToFile ? new StreamWriter(output.OutputFile.Path, false, encoding) : null)
+                using (var writer = output.OutputToFile ? new JsonTextWriter(fileWriter) : new JTokenWriter() as JsonWriter)
+                {
+                    writer.Formatting = Newtonsoft.Json.Formatting.Indented;
+                    writer.Culture = culture;
+
+                    // start array
+                    await writer.WriteStartArrayAsync();
```
Comment (narnil) [inline Frends.Community.Oracle.Query/Extensions.cs]: cancellationToken should be passed to async operations.

File: Frends.Community.Oracle.Query/QueryTask.cs [hunk 4]
```diff
@@ -14,54 +14,50 @@ public class QueryTask
...
         {
             try
             {
-                using (var connection = new OracleConnection(database.ConnectionString))
+                using (var c = new OracleConnection(connection.ConnectionString))
                 {
                     try
                     {
-                        connection.Open();
+                        await c.OpenAsync();
```
Comment (narnil) [inline Frends.Community.Oracle.Query/QueryTask.cs]: Pass cancellationToken.

**Review by narnil — CHANGES_REQUESTED** [review]

**Comment by hodorsoft** [comment]
Requested changes have been made to the same branch

**Pull request merged**